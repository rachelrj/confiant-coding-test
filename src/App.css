.App {
  text-align: center;
  width: 480px;
  margin: 50px auto;
}