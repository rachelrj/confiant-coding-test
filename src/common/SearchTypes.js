const SearchTypes = [
  'Javascript',
  'CSS',
  'HTML',
];

export default SearchTypes;
