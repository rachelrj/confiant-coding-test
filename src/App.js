import React from 'react';
import './App.css';
import '@fontsource/roboto';
import {AppBar, Tabs, Tab} from '@material-ui/core';
import TabPanel from './components/TabPanel/TabPanel';
import SearchBar from './components/SearchBar/SearchBar';
import ResultTable from './components/ResultTable/ResultTable';
import SearchTypes from './common/SearchTypes';

function App() {
  const [searchDetails, setSearchDetails] = React.useState({
    value: 0,
    results: {},
  });

  function a11yProps(index) {
    return {
      'id': `simple-tab-${index}`,
      'aria-controls': `simple-tabpanel-${index}`,
    };
  }
  const handleChange = (event, newValue) => {
    setSearchDetails({
      value: newValue,
      results: {},
    });
  };
  const showResults = (requestResults) => {
    if (requestResults.data && requestResults.data.items) {
      if (requestResults.data.items.length > 0) {
        setSearchDetails({
          results: requestResults.data.items,
          value: searchDetails.value,
        });
      } else {
        setSearchDetails({
          results: {
            'error': 'No results to show.',
          },
          value: searchDetails.value,
        });
      }
    } else {
      setSearchDetails({
        results: {
          'error': 'There was an error encountered.',
        },
        value: searchDetails.value,
      });
    }
  };

  const resultsFound = searchDetails.results &&
    searchDetails.results.length > 0;

  return (
    <div className="App">
      <AppBar position="static">
        <Tabs
          value={searchDetails.value}
          onChange={handleChange}
          aria-label='Three tabs for search: javascript, html, & css.'
        >
          {SearchTypes.map((name, index) =>
            <Tab
              className={'tabLabel'}
              key={index}
              label={SearchTypes[index]}
              {...a11yProps(index)}
            />)}
        </Tabs>
      </AppBar>
      {SearchTypes.map((name, index) =>
        <TabPanel
          key={index}
          value={searchDetails.value}
          index={index}>
          <SearchBar value={searchDetails.value} showResults={showResults}/>
        </TabPanel>,
      )}
      {
        resultsFound && <ResultTable results={searchDetails.results}/>
      }
      {
        !resultsFound && searchDetails.results.error
      }
    </div>
  );
}

export default App;
