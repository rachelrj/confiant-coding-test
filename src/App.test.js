import App from './App';
import SearchTypes from "./common/SearchTypes";
import React from 'react';
import renderer from 'react-test-renderer';
import {act} from 'react-test-renderer';
import {Tabs, Tab} from '@material-ui/core';

test('Renders exactly as many tabs as in SearchTypes arr', () => {
    const component = renderer.create(
      <App />,
    );
    const expectedNumTabLabels = SearchTypes.length;
    expect(component.root.findAllByType(Tabs).length).toEqual(1);
    expect(component.root.findAllByType(Tab).length).toEqual(expectedNumTabLabels);
});

test('Renders tab for each value in SearchTypes arr', () => {
  const component = renderer.create(
    <App />,
  );
  SearchTypes.forEach((value, index) => {
    expect(component.root.findByProps({ label: value })).toBeTruthy();
  });
});

test('Renders Javascript tab on initial load', () => {
  const component = renderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders CSS tab on click', () => {
  const component = renderer.create(
    <App />,
  );

  const button = component.root.findAllByType("button")[1];
  act(button.props.onClick);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders HTML tab on click', () => {
  const component = renderer.create(
    <App />,
  );

  const button = component.root.findAllByType("button")[2];
  act(button.props.onClick);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});