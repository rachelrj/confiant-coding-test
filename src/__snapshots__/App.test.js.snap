// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CSS tab on click 1`] = `
<div
  className="App"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="Three tabs for search: javascript, html, & css."
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls="simple-tabpanel-0"
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tabLabel"
            disabled={false}
            id="simple-tab-0"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Javascript
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="simple-tabpanel-1"
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tabLabel Mui-selected"
            disabled={false}
            id="simple-tab-1"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              CSS
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="simple-tabpanel-2"
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tabLabel"
            disabled={false}
            id="simple-tab-2"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              HTML
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          className="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
          style={
            Object {
              "left": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </header>
  <div
    aria-labelledby="simple-tab-0"
    hidden={true}
    id="simple-tabpanel-0"
    role="tabpanel"
  />
  <div
    aria-labelledby="simple-tab-1"
    hidden={false}
    id="simple-tabpanel-1"
    role="tabpanel"
  >
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <form
        autoComplete="off"
        noValidate={true}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="outlined-basic"
            id="outlined-basic-label"
          >
            Search CSS
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              data-testid="content-input"
              disabled={false}
              id="outlined-basic"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
          
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeLarge"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            Submit
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
  <div
    aria-labelledby="simple-tab-2"
    hidden={true}
    id="simple-tabpanel-2"
    role="tabpanel"
  />
</div>
`;

exports[`Renders HTML tab on click 1`] = `
<div
  className="App"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="Three tabs for search: javascript, html, & css."
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls="simple-tabpanel-0"
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tabLabel"
            disabled={false}
            id="simple-tab-0"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Javascript
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="simple-tabpanel-1"
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tabLabel"
            disabled={false}
            id="simple-tab-1"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              CSS
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="simple-tabpanel-2"
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tabLabel Mui-selected"
            disabled={false}
            id="simple-tab-2"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              HTML
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          className="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
          style={
            Object {
              "left": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </header>
  <div
    aria-labelledby="simple-tab-0"
    hidden={true}
    id="simple-tabpanel-0"
    role="tabpanel"
  />
  <div
    aria-labelledby="simple-tab-1"
    hidden={true}
    id="simple-tabpanel-1"
    role="tabpanel"
  />
  <div
    aria-labelledby="simple-tab-2"
    hidden={false}
    id="simple-tabpanel-2"
    role="tabpanel"
  >
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <form
        autoComplete="off"
        noValidate={true}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="outlined-basic"
            id="outlined-basic-label"
          >
            Search HTML
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              data-testid="content-input"
              disabled={false}
              id="outlined-basic"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
          
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeLarge"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            Submit
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Renders Javascript tab on initial load 1`] = `
<div
  className="App"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="Three tabs for search: javascript, html, & css."
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls="simple-tabpanel-0"
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tabLabel Mui-selected"
            disabled={false}
            id="simple-tab-0"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Javascript
            </span>
            <span
              className="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-controls="simple-tabpanel-1"
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tabLabel"
            disabled={false}
            id="simple-tab-1"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              CSS
            </span>
          </button>
          <button
            aria-controls="simple-tabpanel-2"
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tabLabel"
            disabled={false}
            id="simple-tab-2"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              HTML
            </span>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    aria-labelledby="simple-tab-0"
    hidden={false}
    id="simple-tabpanel-0"
    role="tabpanel"
  >
    <div
      className="MuiBox-root MuiBox-root-7"
    >
      <form
        autoComplete="off"
        noValidate={true}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="outlined-basic"
            id="outlined-basic-label"
          >
            Search Javascript
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              data-testid="content-input"
              disabled={false}
              id="outlined-basic"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
          
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeLarge"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            Submit
          </span>
        </button>
      </form>
    </div>
  </div>
  <div
    aria-labelledby="simple-tab-1"
    hidden={true}
    id="simple-tabpanel-1"
    role="tabpanel"
  />
  <div
    aria-labelledby="simple-tab-2"
    hidden={true}
    id="simple-tabpanel-2"
    role="tabpanel"
  />
</div>
`;
